### ISSUE TITLE
[BUG] Issue running opentelemerty code - `/api/2.0/mlflow/traces` endpoint not found

### ISSUE THREAD

**Issue by laffeychris** [mlflow/mlflow#12380]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

- Client: 2.13.2
- Tracking server: 2.6.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: mac / linux
- **Python version**: 3.11
- **yarn version, if running the dev UI**: N/A

### Describe the problem

Expected: Code that uses opentelemetry (e.g. snowflake) runs correctly 

Observed: Error (stack trace shown below) is raised. 

I believe it's coming from version `2.13.2`, where databricks tracing has been enabled [here](https://mlflow.org/docs/latest/llms/tracing/index.html#tracing-client-apis)

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
import mlflow
from snowflake.snowpark import Session

MLFLOW_TRACKING_URI = "https://mlflow.dev.company.com/"
EXPERIMENT_NAME = "test-model"

connection_parameters = {}

if __name__ == "__main__":
    mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)
    mlflow.set_experiment(EXPERIMENT_NAME)
    model = mlflow.pyfunc.load_model(model_uri="models:/{}/latest".format(EXPERIMENT_NAME))

    session = Session.builder.configs(connection_parameters).create()
    exists_df = session.sql("SHOW TABLES LIKE 'TEST'").collect()
```

### Stack trace

```
Traceback (most recent call last):
  File "***/test.py", line 24, in <module>
    exists_df = session.sql("SHOW TABLES LIKE 'TEST'").collect()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/snowflake/snowpark/_internal/telemetry.py", line 145, in wrap
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/snowflake/snowpark/dataframe.py", line 596, in collect
    with open_telemetry_context_manager(self.collect, self):
  File "/opt/homebrew/Cellar/python@3.11/3.11.9/Frameworks/Python.framework/Versions/3.11/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/snowflake/snowpark/_internal/open_telemetry.py", line 38, in open_telemetry_context_manager
    with tracer.start_as_current_span(name) as cur_span:
  File "*****/.venv/lib/python3.11/site-packages/opentelemetry/util/_decorator.py", line 58, in __enter__
    return next(self.gen)  # type: ignore
           ^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/opentelemetry/sdk/trace/__init__.py", line 1075, in start_as_current_span
    span = self.start_span(
           ^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/opentelemetry/sdk/trace/__init__.py", line 1165, in start_span
    span.start(start_time=start_time, parent_context=context)
  File "*****/.venv/lib/python3.11/site-packages/opentelemetry/sdk/trace/__init__.py", line 922, in start
    self._span_processor.on_start(self, parent_context=parent_context)
  File "*****/.venv/lib/python3.11/site-packages/opentelemetry/sdk/trace/__init__.py", line 164, in on_start
    sp.on_start(span, parent_context=parent_context)
  File "*****/.venv/lib/python3.11/site-packages/mlflow/tracing/processor/mlflow.py", line 70, in on_start
    trace_info = self._start_trace(span)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/tracing/processor/mlflow.py", line 122, in _start_trace
    return self._client._start_tracked_trace(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/tracking/client.py", line 946, in _start_tracked_trace
    return self._tracking_client.start_trace(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/tracking/_tracking_service/client.py", line 193, in start_trace
    return self.store.start_trace(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/store/tracking/rest_store.py", line 252, in start_trace
    response_proto = self._call_endpoint(StartTrace, req_body)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/store/tracking/rest_store.py", line 81, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/utils/rest_utils.py", line 304, in call_endpoint
    response = verify_rest_response(response, endpoint)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "*****/.venv/lib/python3.11/site-packages/mlflow/utils/rest_utils.py", line 180, in verify_rest_response
    raise MlflowException(
mlflow.exceptions.MlflowException: API request to endpoint /api/2.0/mlflow/traces failed with error code 404 != 200. Response body: '<!doctype html>
<html lang=en>
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
'
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@laffeychris thanks for reporting this! we'll fix this soon :)

**Comment by B-Step62** [comment]
Fixed in #12457. It will be available in the next patch release.
Than you so much for raising the issue, @laffeychris!
